packages/grid/_modules_/grid/models/rows.ts

```diff
@@ -7,6 +7,11 @@ export interface RowModel extends ObjectWithId {
   [key: string]: any;
 }
 
+export type GridUpdateAction = 'delete' | 'upsert';
```

Comment:
How about we only support delete? It work with delete because the row is no longer in the store. With upsert, I don't think that it makes sense to have an imperative action stored into a state.

```suggestion
export type GridUpdateAction = 'delete';
```

Reply:
upsert is the default behavior.
So if it's not a delete action then it upsert. I put that as sometimes developer want to fill all fields, ie backend guys

Reply:
Hum, my concern is with the leaky abstraction. Seeing rows with `_action: 'upsert'` is a signal that the abstraction is wrong.

Reply:
Personally, I prefer flat structure. I think it's easier to manipulate from a server perspective, I get an array of updates and call updateRows, no need to map the array...
If you map the array, you could just call updateRows and deleteRows...
🤔

Reply:
The server drives the UI without the need for mapping.

```tsx
.subscribe((rowUpdates) => {
  apiRef.current.updateRows(rowUpdates);
});

```

Reply:
It's definitely a tradeoff. Oh yeah, maybe a `deleteRows` might actually be all we need for streaming support :p.

Reply:
Well it defaults to upsert so backend guys might hook the grid to a pricing feed that will result in 
```tsx
apiRef.current.updateRows({id: 1, price: 123});
```

Then the user filters the grid and don't want to see some rows, the server will just send some other updates to the stream to delete the rows that should not be visible, so it will just need to add the` _action: delete`
```tsx
 apiRef.current.updateRows({id: 1, _action: delete});
```
So it can use the same channel to completely manipulate the grid rows.

Reply:
@dtassone Are you making a point in favor of my proposed diff? :)

```diff
-export type GridUpdateAction = 'delete' | 'upsert';
+export type GridUpdateAction = 'delete';
```

Reply:
Fine. Let's remove it ;)